modules/Products/models/Record.php

```diff
@@ -138,50 +142,43 @@ public function getBaseCurrencyDetails()
 	 * Static Function to get the list of records matching the search key.
 	 *
 	 * @param string $searchKey
+	 * @param mixed  $moduleName
+	 * @param mixed  $limit
+	 * @param mixed  $operator
 	 *
 	 * @return <Array> - List of Vtiger_Record_Model or Module Specific Record Model instances
 	 */
 	public static function getSearchResult($searchKey, $moduleName = false, $limit = false, $operator = false)
 	{
 		$query = false;
-		if (false !== $moduleName && ('Products' == $moduleName || 'Services' == $moduleName)) {
-			$currentUser = \Users_Record_Model::getCurrentUserModel();
-			$adb = \PearDatabase::getInstance();
-			$params = ['%' . $currentUser->getId() . '%', "%$searchKey%"];
-			$queryFrom = 'SELECT u_yf_crmentity_search_label.`crmid`,u_yf_crmentity_search_label.`setype`,u_yf_crmentity_search_label.`searchlabel` FROM `u_yf_crmentity_search_label`';
-			$queryWhere = ' WHERE u_yf_crmentity_search_label.`userid` LIKE ? && u_yf_crmentity_search_label.`searchlabel` LIKE ?';
-			$orderWhere = '';
+		if (false !== $moduleName && ('Products' === $moduleName || 'Services' === $moduleName)) {
+			$currentUserId = \Users_Record_Model::getCurrentUserModel()->getId();
+			$query = (new App\Db\Query())->select(['u_#__crmentity_search_label.crmid', 'u_#__crmentity_search_label.setype', 'u_#__crmentity_search_label.searchlabel'])
+				->from('u_#__crmentity_search_label')
+				->where(['and', ['like', 'u_#__crmentity_search_label.userid', ",{$currentUserId},"], ['like', 'u_#__crmentity_search_label.searchlabel', $searchKey]]);
 			if (false !== $moduleName) {
-				$multiMode = is_array($moduleName);
-				if ($multiMode) {
-					$queryWhere .= sprintf(' AND u_yf_crmentity_search_label.`setype` IN (%s)', $adb->generateQuestionMarks($moduleName));
-					$params = array_merge($params, $moduleName);
+				if (is_array($moduleName)) {
```

Comment:
is_array to remove
$query->andWhere(['u_#__crmentity_search_label.setype' => $moduleName]);